docs/language-service.md

```diff
@@ -0,0 +1,38 @@
+# Key features of the Bicep language service
+
+The bicep language service (implemented in the [bicep VS Code extension](./installing.md#install-the-bicep-vs-code-extension)) is capable of many of the features you would expect out of other language tooling. Here's all of the features that are currently implemented.
+
+## Validation and type checking
+
+The bicep compiler has a type system that ensure your code is authored correctly. We will check the basic types that are built into the bicep language, as well as types of returned functions, and types of resources. Depending on the type of validation, you will see either a warning in yellow which will successfully compile with `bicep build` or you will see an error in red which will fail to compile.
```

Comment:
would like to expand on this more, but don't feel that I'm using the right language. @anthony-c-martin / @majastrz can you take a closer look at this

Reply:
Maybe change the section title to just "Validation" and use the following paragraph to start with? We could potentially list all the things we validate, but we'll have to maintain the list because it will grow over time.

The bicep compiler validates that your code is authored correctly. We always validate the syntax of each file and whenever possible also validate the return types of all expressions (functions, resource bodies, parameters, outputs, etc.). Depending on the type of validation, you will see either a warning in yellow which will successfully compile with `bicep build` or you will see an error in red which will fail to compile.

Reply:
like it - thanks

Reply:
Btw, I will have a much more detailed writeup when I finish docs for #36.